src/app/components/table.tsx

```diff
@@ -0,0 +1,177 @@
+import { useMemo } from 'react';
+
+type DataNode = LeafNode | ParentNode;
+type LeafNode = {
+  name: string;
+  value: number;
+};
+type ParentNode = {
+  name: string;
+  children: DataNode[];
+};
```

Comment:
Is there a difference between this type and the SunburstData type? Feel free to move that out into a model file and rename it if they are the same.